src/drivers/imu/invensense/icm42688p/ICM42688P.cpp

```diff
@@ -589,24 +561,100 @@ void ICM42688P::FIFOReset()
 	_drdy_fifo_read_samples.store(0);
 }
 
+static constexpr int32_t reassemble_20bit(const uint32_t a, const uint32_t b, const uint32_t c)
+{
+	// 0xXXXAABBC
+	uint32_t high   = ((a << 12) & 0x000FF000);
+	uint32_t low    = ((b << 4)  & 0x00000FF0);
+	uint32_t lowest = (c         & 0x0000000F);
+
+	uint32_t x = high + low + lowest;
+
+	if (a & Bit7) {
+		// sign extend
+		x = 0xFFF00000u + x;
```

Comment:
Using bit ops might be a bit faster:
```suggestion
	uint32_t x = high | low | lowest;

	if (a & Bit7) {
		// sign extend
		x |= 0xFFF00000u;
```